src/Columns/ColumnObject.cpp

```diff
@@ -711,70 +736,49 @@ void ColumnObject::serializePathAndValueIntoArena(DB::Arena & arena, const char
 const char * ColumnObject::deserializeAndInsertFromArena(const char * pos)
 {
     size_t current_size = size();
-    /// Deserialize paths and values and insert them into typed paths, dynamic paths or shared data.
-    /// Serialized paths could be unsorted, so we will have to sort all paths that will be inserted into shared data.
-    std::vector<std::pair<std::string_view, std::string_view>> paths_and_values_for_shared_data;
+    /// First deserialize typed paths. They come first.
+    for (auto path : sorted_typed_paths)
+        pos = typed_paths.find(path)->second->deserializeAndInsertFromArena(pos);
+
+    /// Second deserialize all other paths and values and insert them into dynamic paths or shared data.
     auto num_paths = unalignedLoad<size_t>(pos);
     pos += sizeof(size_t);
+    const auto [shared_data_paths, shared_data_values] = getSharedDataPathsAndValues();
     for (size_t i = 0; i != num_paths; ++i)
     {
         auto path_size = unalignedLoad<size_t>(pos);
         pos += sizeof(size_t);
         std::string_view path(pos, path_size);
```

Comment:
Shouldn't we have some sanity check here? Can we read such column from client, for example or we validated it beforehand? Anyway, if it's possible to check size with maximal it's worth doing at least to protect from some bug for example

Reply:
> Can we read such column from client, for example or we validated it beforehand?

AFAIK methods `serializeValueIntoArena/serializePathAndValueIntoArena` are used only internally in aggregation and the data cannot be serialized/deserialized from outside (we can have serialized aggregate functions states, but it's different). And I don't see any similar checks in other columns implementation.

If you have good suggestion for max size to check, we can add it